backends/p4fmt/attach.cpp

```diff
@@ -0,0 +1,71 @@
+#include "backends/p4fmt/attach.h"
+
+#include "frontends/common/parser_options.h"
+#include "lib/source_file.h"
+
+namespace P4::P4Fmt {
+
+void Attach::addPrefixComments(NodeId node, const Util::Comment *prefix) {
+    commentsMap[node].prefix.push_back(prefix);
+}
+
+void Attach::addSuffixComments(NodeId node, const Util::Comment *suffix) {
+    commentsMap[node].suffix.push_back(suffix);
+}
+
+const Attach::CommentsMap &Attach::getCommentsMap() const { return commentsMap; }
+
+const IR::Node *Attach::attachCommentsToNode(IR::Node *node, TraversalType ttype) {
+    if (node == nullptr || !node->srcInfo.isValid() || processedComments.empty()) {
+        return node;
+    }
+
+    std::filesystem::path sourceFile(node->srcInfo.getSourceFile().string_view());
+    if (isSystemFile(sourceFile.string())) {
```

Comment:
`sourceFile` is entirely unused below. What is the purpose of this useless `cstring => std::filesystem::path => std::string => cstring` conversion here? Why can't you simply do `if (isSystemFile(node->srcInfo.getSourceFile())`?